cluster/ssl_config.go

```diff
@@ -28,38 +28,39 @@ import (
 )
 
 // SSLConfig is SSL configuration for client.
-// SSLConfig has tls.Config embedded in it so that users can set any field
-// of tls config as they wish. SSL config also has some helpers such as SetCaPath, AddClientCertAndKeyPath to
-// make configuration easier for users.
+// SSLConfig has tls.Config embedded in it so that users can set any field of tls config as they wish.
 type SSLConfig struct {
-	TLSConfig *tls.Config `json:"-"`
-	Enabled   bool
-}
-
-func NewSSLConfig() SSLConfig {
-	return SSLConfig{TLSConfig: &tls.Config{}}
+	tlsConfig *tls.Config
```

Comment:
Will the users are able to set TLS config via JSON somehow?

Reply:
There are a few fields in `tls.Config` that  could be set via JSON, but most of it cannot. Ideally setting `config.Cluster.SSL.Enabled = true` without customizing `tlsConfig` should be enough for most users.

Reply:
I see. I find it valuable to configure at least client and CA certificates via JSON, as I assume it will be used this way when the users want to use self-signed certificates but I don't know what we can do about it. So, I am resolving this.